fvm/src/syscalls/mod.rs

```diff
@@ -62,7 +61,10 @@ pub fn update_gas_available(
     Ok(())
 }
 
-pub fn charge_for_exec(
+/// Applies gas charges upon passing control to the FVM via a syscall.
+/// It first applies accumulated execution gas, and then applies the syscall
+/// charge for the currently active syscall.
+pub fn apply_charges_on_syscall(
```

Comment:
Being nitpicky here, but _technically_ this is a charge on exiting wasm, not just syscalls.

Reply:
This is actually a bug. We call this when we return from wasm as well.

Reply:
Good catch, thanks for that!

Reply:
Will fix this tomorrow.